client/src/components/UpdateUser/UpdateUser.js

```diff
@@ -0,0 +1,179 @@
+            this.handleChange({
+                target: {
+                    name,
+                    value: this.state[name]
+                }
+            });
+        }
+        for (let message of Object.values(errors)) {
+            if (message.length > 0) valid = false;
+        }
+        this.setState({validated: true});
+        return valid;
+    }
+
+    handleSubmit(event) {
+        const { onSubmit } = this.props;
+        const { errors, email, firstName, lastName } = this.state;
+        event.preventDefault();
+        if (this.validateForm(errors)) {
+            onSubmit({email, firstName, lastName});
+        }
+    }
+
+    handleChange(event) {
+        const { errors } = this.state;
+        const { name, value } = event.target;
+        if (event.preventDefault) event.preventDefault();
+
+        switch (name) {
+            case 'firstName':
+                errors.firstName = value
+                    ? ''
+                    : 'First name is required';
+                break;
+            case 'lastName':
+                errors.lastName = value
+                    ? ''
+                    : 'Last name is required';
+                break;
+            case 'email':
+                errors.email = validEmailRegex.test(value)
+                    ? ''
+                    : 'Email is not valid';
+                break;
+            default:
+                break;
+        }
+
+        this.setState({errors, [name]: value})
+    }
+
+    render() {
+        const { email, firstName, lastName, errors, validated } = this.state;
+        const { error, success, needsConfirmation, history, user } = this.props;
+
+        const hasChanges = (email !== user.email) || (firstName !== user.firstName) || (lastName !== user.lastName);
+
+        return (
+            <div className="d-flex flex-column align-items-center">
+                <Card className="mb-4">
+                    <Card.Header>
+                        <Card.Title>Update Your Information</Card.Title>
+                    </Card.Header>
+                    <Card.Body>
+                        {!success &&
+                            <Form className="update-account-form" noValidate onSubmit={event => this.handleSubmit(event)}>
+                                <Form.Group>
+                                    <Form.Label>First Name</Form.Label>
+                                    <Form.Control
+                                        type="text"
+                                        name="firstName"
+                                        value={firstName}
+                                        onChange={event => this.handleChange(event)}
+                                        className="text-control"
+                                        required
+                                        noValidate
+                                        isInvalid={errors.firstName && validated}
```

Comment:
Is there are good `autoComplete` prop for this?

Reply:
No, there's no `new-username` unfortunately: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

Reply:
What about `given-name` or just `name`?